### PR TITLE
Add files.get command and tests

### PR DESCRIPTION
**Pull request by noffle**
Builds on @nginnever's work, bringing the rest of the implementation to completion.

Requires https://github.com/ipfs/interface-ipfs-core/pull/28 to be merged first to get tests passing.

### CODE REVIEW

**Comment by diasdavid** [comment]
@noffle I got a:

```
 1) .files .get with a multihash:
     Uncaught AssertionError: expected 'QmPZ9gcCEpqKTo6aq61g2nXGUhM4iCL3ewB6LDXZCtioEB' to deeply equal <Buffer 12 20 12 0f 6a f6 01 d4 6e 10 b2 d2 e1 1e d7 1c 55 d2 5f 30 42 c2 25 01 e4 1d 12 46 e7 a1 e9 d3 d8 ec>
```

Seems that on the interface-ipfs-core tests expect two different result types, check:

- https://github.com/ipfs/interface-ipfs-core/pull/28/files#r73695467
- https://github.com/ipfs/interface-ipfs-core/pull/28/files#r73695554

**Comment by noffle** [comment]
Addressed! See https://github.com/ipfs/interface-ipfs-core/pull/28#issuecomment-237993885

File: test/api/get.spec.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
+/* eslint-env mocha */
```
Comment (diasdavid) [inline test/api/get.spec.js]: Do these tests cover something that interface-ipfs-core tests don't cover? If yes, why not have all of them at interface-ipfs-core for get?

Comment (diasdavid) [inline test/api/get.spec.js]: I'm also getting am error from this file, which makes all the browser tests to fail:

```
ERROR in ./test/api/get.spec.js
Module build failed: TypeError: Cannot read property 'apply' of undefined
```

File: test/api/get.spec.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+      streamEqual(res, testfileBig, (err, equal) => {
+        expect(err).to.not.exist
+        expect(equal).to.be.true
+        done()
+      })
+    })
+  })
+
+  describe('promise', () => {
+    it.skip('get', (done) => {
```
Comment (diasdavid) [inline test/api/get.spec.js]: should this one be skipped?

File: test/api/get.spec.js (hunk 1 above)
Comment (diasdavid) [inline test/api/get.spec.js]: I think I get it, these are the tests for the options supported by go-ipfs

File: test/api/get.spec.js [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+let testfile
+let testfileBig
+
+if (isNode) {
+  testfile = fs.readFileSync(path.join(__dirname, '/../testfile.txt'))
+  testfileBig = fs.createReadStream(path.join(__dirname, '/../15mb.random'), { bufferSize: 128 })
+} else {
+  testfile = require('raw!../testfile.txt')
+}
+
```
Comment (diasdavid) [inline test/api/get.spec.js]: Add a comment here to explain that these are the tests for the options supported by go-ipfs, not yet provided in js-ipfs.

File: test/api/get.spec.js [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+const extract = require('tar-stream').extract
+
+let testfile
+let testfileBig
+
+if (isNode) {
+  testfile = fs.readFileSync(path.join(__dirname, '/../testfile.txt'))
+  testfileBig = fs.createReadStream(path.join(__dirname, '/../15mb.random'), { bufferSize: 128 })
+} else {
+  testfile = require('raw!../testfile.txt')
```
Comment (diasdavid) [inline test/api/get.spec.js]: We don't use the raw buffer loader anymore, just brfs

Comment (diasdavid) [inline test/api/get.spec.js]: Meaning, we can use line 21 for both Node.js and browser

File: test/api/get.spec.js [hunk 5]
```diff
@@ -0,0 +1,125 @@
+/* eslint-env mocha */
+/* globals apiClients */
+'use strict'
+
+const expect = require('chai').expect
+const isNode = require('detect-node')
+const fs = require('fs')
+const bl = require('bl')
```
Comment (diasdavid) [inline test/api/get.spec.js]: not used in this file

File: test/api/get.spec.js [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+const concat = require('concat-stream')
+const through = require('through2')
+
+const path = require('path')
+const streamEqual = require('stream-equal')
+
+const extract = require('tar-stream').extract
```
Comment (diasdavid) [inline test/api/get.spec.js]: not used

**Comment by diasdavid** [comment]
Fixed the browser tests on https://github.com/ipfs/js-ipfs-api/tree/feat/ipfs.files.get :)

**Comment by noffle** [comment]
Closed in favour of https://github.com/ipfs/js-ipfs-api/pull/337.

**Pull request closed without merging**